modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java

```diff
@@ -135,4 +149,34 @@ private String unwrapChildKey(String key)
     {
         return key.replaceAll(CHILD_ELEMENT_KEY_PREFIX, "").replaceAll(CHILD_ELEMENT_KEY_SUFFIX, "");
     }
+
+    private void checkParameterGroupExclusivenessForModel(EnrichableModel model, Set<String> resolverKeys) throws ConfigurationException
+    {
+        Optional<ParameterGroupModelProperty> parameterGroupModelProperty = model.getModelProperty(ParameterGroupModelProperty.class).filter(mp -> mp.hasExclusion());
+        if (parameterGroupModelProperty.isPresent())
+        {
+            checkParameterGroupExclusiveness(model, parameterGroupModelProperty.get(), resolverKeys);
+        }
+    }
+
+    private void checkParameterGroupExclusiveness(EnrichableModel model, ParameterGroupModelProperty parameterGroupModelProperty, Set<String> resolverKeys) throws ConfigurationException
+    {
+        for (ParameterGroup<?> group : parameterGroupModelProperty.getGroups())
+        {
+            Set<String> parametersFromGroup = group.getOptionalParameters().stream()
+                    .map(p -> p.getName())
+                    .filter(name -> resolverKeys.contains(name))
+                    .collect(toSet());
+
+            if (parametersFromGroup.size() > 1)
+            {
+                throw new ConfigurationException(createStaticMessage(format("In %s '%s', parameter group '%s' has been set with the following parameters [%s] but only one should be set at any time due to its exclusive relation", getComponentModelTypeName(model), getModelName(model), group.getType().getName(), Joiner.on(", ").join(parametersFromGroup))));
```

Comment:
bad error message. The extension user does not know about parameter groups. This should just say that parameters x, y, z cannot be set at the same time. 

Also split this message creation into several lines.